CLI exits on first command when a credential is invalid.

tomohikokimura opened the pull request:
Currently CLI goes back to a normal command line loop showing a
prompt even when a previous command failed because of an invalid
credential. This is confusing as it can appear as if authentication
was successful and CLI was ready to take commands. The new behavior
is that CLI will exit upon a first command if a credential is
invalid, displaying a corresponding error message.

Fix MN-718

Change-Id: I2ceb940f35c7418edee3601a7fc55874e24436dd

Commit:
CLI exits on first command when a credential is invalid.

Currently CLI goes back to a normal command line loop showing a
prompt even when a previous command failed because of an invalid
credential. This is confusing as it can appear as if authentication
was successful and CLI was ready to take commands. The new behavior
is that CLI will exit upon a first command if a credential is
invalid, displaying a corresponding error message.

Fix MN-718

Change-Id: I2ceb940f35c7418edee3601a7fc55874e24436dd

Commit:
CLI exits on first command when a credential is invalid.

Currently CLI goes back to a normal command line loop showing a
prompt even when a previous command failed because of an invalid
credential. This is confusing as it can appear as if authentication
was successful and CLI was ready to take commands. The new behavior
is that CLI will first run a test command to see if a credential is
valid, and exit right after if invalid, displaying a corresponding
error message.

Fix MN-718

Change-Id: I2ceb940f35c7418edee3601a7fc55874e24436dd

tomohikokimura commented:
Hello Rosella;

Could you please review this? This is a trivial change, fixing MN-718.

Thanks,
Tomohiko

tomohikokimura commented:
Hello Rosella,

It's just a friendly ping. Could you take a look at this patch once after you get enough rest to recover from OpenStack Summit, please?

Grazie,
Tomohiko

admytren commented:
Tomohiko, this sounds like a great improvement. What do you think about automatically running a command when user logs in (for example, "list tunnel-zone") so that credentials could be validated immediately?

Commit:
Merge branch 'dev/tomohiko/mn-718/20131031' of https://github.com/midokura/python-midonetclient into dev/tomohiko/mn-718/20131031

tomohikokimura commented:
Thank you for your suggestion, @admytren ! I thought about that, too, but I couldn't come up with an idea about where to do that cleanly. Now that I see where that can be done nicely, I reflected your suggestion to the patch. @rossella or @admytren , can you guys take a look?

admytren commented on src/bin/midonet-cli:
```diff
@@ -3393,7 +3399,9 @@ if __name__ == '__main__':
-            cli.cmdloop()
+            # Test the credential by running a sample command first before
+            # starting a command loop.
+            if not cli.onecmd('list host'): cli.cmdloop()
```
Could you make sure that this command doesn't provide any output on the screen? Since it's just a connectivity check it would be inconvenient to the user to see the output. Additionally, I would recommend listing tunnel zones as that operation may be much lighter. In production environment it's possible to have a lot of hosts.

tomohikokimura replied:
Yes, I checked that this wouldn't output anything on the screen. I also changed the test command to "list tunnel-zone" as per your request.

Commit:
Added an error handling for connection refused to API server.

Change-Id: I4eb640e3072f880491438985872daae425e3296d

Commit:
Added an error handling for connection refused to API server.

Change-Id: I4eb640e3072f880491438985872daae425e3296d

Commit:
Merge branch 'dev/tomohiko/mn-718/20131031' of https://github.com/midokura/python-midonetclient into dev/tomohiko/mn-718/20131031

Conflicts:
	src/bin/midonet-cli

Change-Id: I9b966fd8574bea658e1e679e37bea8c25c6bebbc

tomohikokimura commented:
I changed the sample test command to "list tunnel-zone" as per Artem's suggestion. I also added a handling of connection refused between the API server. Can @admytren, @rossella or anyone take a look at this and merge, please?

Tomohiko

admytren commented:
This improves the usability greatly!
